slotfills/sections/ajax-autocomplete.tsx

```diff
@@ -0,0 +1,247 @@
+import { __ } from '@wordpress/i18n';
+import apiFetch from '@wordpress/api-fetch';
+import { isArray } from 'lodash';
+
+interface AjaxAutocompleteProps {
+  ajaxAction: string;
+  label?: string;
+  initialValue?: string;
+  setValue: Function;
+  showEditLink?: boolean;
+}
+
+interface Post {
+  label: string;
+  value: string;
+}
+
+interface FormDataProps {
+  [key: string]: string;
+}
+
+export default function AjaxAutocomplete({
+  ajaxAction,
+  initialValue, // eslint-disable-line @typescript-eslint/no-unused-vars
+  label,
+  setValue,
+  showEditLink = false,
+}: AjaxAutocompleteProps) {
+  const [foundPosts, setFoundPosts] = useState<Post[]>([]);
+  const [searchText, setSearchText] = useState('');
+  const [working, setWorking] = useState(false);
+  const threshold = 3;
+  const {
+    ajaxurl,
+    fm_search: {
+      nonce: fmSearchNonce = '',
+    } = {},
+  } = (window as any);
+
+  const onChange = (newValue:Post) => {
+    const selectedValue = newValue ? newValue.value : null;
+    setValue(selectedValue);
+  };
+
+  /**
+   * Handles a post selection. Calls `onChange` from props with
+   * the found post object.
+   * @param {object} foundPost - The selected post.
+   */
+  const handlePostSelect = (foundPost: Post) => {
+    // Call the passed onChange function from props with the post object.
+    onChange(foundPost);
+
+    // Reset the internal state.
+    setFoundPosts([]);
+    setSearchText('');
+  };
+
+  useEffect(() => {
+    /**
+     * Loads found posts for the given post type and search text from the API.
+     * @param {string} searchText - The text string to use when searching.
+     */
+    const loadById = (id: number) => {
+      setWorking(true);
+
+      apiFetch({
+        path: `/wp/v2/search?include=${id}`,
+      })
+        .then((result) => {
+          if (!isArray(result)) {
+            return;
+          }
+          if (result.length === 0) {
+            return;
+          }
+          setSearchText(result[0].title);
+
+          setWorking(false);
+        });
+    };
+    if (!Number.isNaN(parseInt(initialValue, 10))) {
+      loadById(parseInt(initialValue, 10));
+    }
+  }, [initialValue]);
+
+  /**
+   * Loads found posts for the given post type and search text from the API.
+   * @param {string} searchText - The text string to use when searching.
+   */
+  const loadFoundPosts = () => {
+    // If the search text is not at the threshold, bail.
+    if (threshold > searchText.length) {
+      return;
+    }
+
+    setWorking(true);
+
+    const formdata:FormDataProps = {
+      action: ajaxAction,
+      fm_context: 'post', // TODO: make this dynamic
+      fm_subcontext: 'demo-autocomplete', // TODO: make this dynamic
+      fm_autocomplete_search: searchText,
+      fm_search_nonce: fmSearchNonce,
+      fm_custom_args: null,
+    };
+
+    const str = Object.keys(formdata).map((key) => (
+      `${key}=${formdata[key]}`
+    )).join('&');
+
+    fetch(ajaxurl, {
```

Comment:
Why was `apiFetch` not used here?

Reply:
I forgot that `apiFetch` supports `url` to pass a url. I was only thinking of `path` which gets appended to `/wp-json`